operator/main.py

```diff
@@ -464,22 +466,178 @@ def handle_modified(core_v1_client, obj):
+        template_args["host_brick_path"] = storage_unit.path
+        template_args["kube_hostname"] = storage_unit.kube_host
+        template_args["serverpod_name"] = get_brick_hostname(
+            volname,
+            idx,
+            suffix=False
+        )
+        template_args["brick_path"] = "/bricks/%s/data/brick" % volname
+        template_args["brick_index"] = idx
+        template_args["brick_device"] = storage_unit.device
+        template_args["pvc_name"] = storage_unit.pvc
+        template_args["brick_device_dir"] = storage_unit.brick_device_dir
+        template_args["brick_node_id"] = storage_unit.node_id
+        template_args["k8s_dist"] = K8S_DIST
+
+        filename = os.path.join(MANIFESTS_DIR, "server.yaml")
+        template(filename, **template_args)
+        lib_execute(KUBECTL_CMD, DELETE_CMD, "-f", filename)
+        logging.info(logf(
+            "Deleted Server pod",
+                volname=volname,
+                manifest=filename,
+                node=storage_unit.node
+        ))
+
+def get_storages():
+    """Parse list of storages from
+       JSON input and get storages
+    """
+
+    cmd = [
+        "kubectl", "get", "configmap",
+        "kadalu-info", "-nkadalu", "-ojson"
+    ]
+    try:
+        resp = utils_execute(cmd)
+        #list_storages does not make use of use 'args' thus pass None
+        storages = list_storages(resp.stdout, None)
+    except CommandError as msg:
+        logging.error(logf(
+            "Failed to get Kadalu ConfigMap",
+            error=msg
+        ))
+
+    return storages
+
+def get_num_pvs(storage):
+    """Get PV count for all listed storages"""
+
+    dbpath = "/bricks/" + storage.storage_name + "/data/brick/stat.db"
+    query = ("select count(pvname) from pv_stats;")
+    cmd = ["kubectl", "exec", "-i",
+            storage.storage_units[0].podname,
+            "-c", "glusterfsd", "-nkadalu", "--", "sqlite3",
+            dbpath,
+            query
+    ]
+
+    try:
+        resp = utils_execute(cmd)
+        parts = resp.stdout.strip().split("\n")
+        storage.total_size_bytes = float(parts[0].strip())
+        pv_stats_parts = parts[1].strip().split("|")
+        storage.pv_count = int(pv_stats_parts[0])
+
+    except CommandError as msg:
+        logging.error(logf(
+            "Failed to get size details of the "
+            "storage \"%s\"" % storage.storage_name,
+            error=msg
+        ))
+
+def handle_deleted(core_v1_client, obj):
     """
     If number of pvs provisioned from that volume
     is zero - Delete the respective server pods
     If number of pvs is not zero, wait or periodically
     check for num_pvs. Delete Server pods only when pvs becomes zero.
     """
-    # TODO
+
+    volname = obj["metadata"]["name"]
+
     logging.warning(logf(
-        "DELETED handle called, but not implemented"
+        "Delete requested",
+        volname=volname
     ))
 
+    # Intialise Variables
+    storage = Storage()
```

Comment:
this gets overwritten.. is that OK?

Reply:
I was not able test it properly since the PV was not binding. Ill change the default values and check again for storage.pv_count